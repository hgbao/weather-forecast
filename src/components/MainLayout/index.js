import { Layout as AntdLayout } from 'antd';
import PropTypes from 'prop-types';

import PageTitle from '@/components/PageTitle';

import styles from './styles.module.less';

const MainLayout = ({ children, title }) => {
  return (
    <AntdLayout>
      <PageTitle title={title} />

      <AntdLayout className={styles.container}>
        <main>{children}</main>
      </AntdLayout>
    </AntdLayout>
  );
};

MainLayout.propTypes = {
  children: PropTypes.node.isRequired,
  title: PropTypes.string.isRequired,
};

export default MainLayout;
