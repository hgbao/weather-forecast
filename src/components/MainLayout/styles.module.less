.container {
  display: flex;
  min-height: 100vh;

  main {
    flex: 1;
    padding: 16px 32px;
    overflow-y: auto;
    background-color: @color-background;
  }
}
