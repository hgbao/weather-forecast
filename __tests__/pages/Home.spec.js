import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import MainLayout from '@/components/MainLayout';
import Home from '@/pages';

describe('Home', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<Home />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render with correct page title', () => {
    const wrapper = shallow(<Home />);
    wrapper.find(MainLayout).prop('title').should.equal('Home');
  });
});
