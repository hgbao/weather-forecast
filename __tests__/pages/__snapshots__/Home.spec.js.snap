// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should match snapshot 1`] = `
<MainLayout
  title="Home"
>
  <Row
    gutter={
      Array [
        16,
        16,
      ]
    }
  >
    <Col
      span={24}
    >
      <RequestSearchBar
        onOptionsRequest={[Function]}
        onSelect={[Function]}
      />
    </Col>
    <Col
      span={24}
    >
      <Empty />
    </Col>
  </Row>
</MainLayout>
`;
